src/main.ts

```diff
@@ -0,0 +1,256 @@
+
+  await exec.exec(
+    '/bin/bash',
+    [
+      '-c',
+      `gh run list -w "${context.workflow}" | grep "${context.workflow}	${headRef}" | grep -v "completed	cancelled" | grep -v "in_progress" | head -n 1 | awk -F" " '{print $1"/"$2}'`,
+    ],
+    options,
+  );
+
+  core.info(`GH Found status: ${lastStatus}`);
+
+  return lastStatus;
+}
+
+/**
+ * Gets the last workflow run status
+ *
+ * @returns last run status
+ */
+async function getLastRunStatus() {
+  let lastStatus = '';
+
+  const cacheKey = await cache.restoreCache(
+    cachePaths,
+    cachePrimaryKey,
+    cacheRestoreKeys,
+  );
+
+  if (cacheKey) {
+    try {
+      lastStatus = await fsp.readFile(cachePaths[0], 'utf8');
+      core.info(`Cache Found status: ${lastStatus}`);
+    } catch (err) {
+      lastStatus = await getStatusFromGithub();
+    }
+  } else {
+    lastStatus = await getStatusFromGithub();
+  }
+
+  return lastStatus.trim();
+}
+
+/**
+ * Prepare slack notification
+ *
+ * @param message contained in the Slack notification
+ * @param status current status to notify
+ * @returns the Slack message body
+ */
+async function prepareSlackNotification(message: string, status: string) {
+  const {
+    runId,
+    workflow,
+    actor,
+    serverUrl,
+    eventName: event,
+    repo: { owner },
+  } = context;
+  const color = status === 'success' ? 'good' : 'danger';
+
+  const headRef = await getHeadRef();
+
+  const messageBody = {
+    icon_emoji: ':bangbang:', // User icon, you can also use custom icons here
+    attachments: [
+      {
+        // this defines the attachment block, allows for better layout usage
+        color: `${color}`, // color of the attachments sidebar.
+        author_name: `${actor}`,
+        author_link: `${serverUrl}/${actor}`,
+        author_icon: `${serverUrl}/${actor}.png?size=32`,
+        fields: [
+          // actual fields
+          {
+            title: 'Repository',
+            value: `${repository}`,
+            short: true,
+          },
+          {
+            title: 'Branch',
+            value: `${headRef}`,
+            short: true,
+          },
+          {
+            title: 'Action URL',
+            value: `<${serverUrl}/${owner}/${repository}/actions/runs/${runId}|${workflow}>`,
+            short: true,
+          },
+          {
+            title: 'Event',
+            value: `${event}`,
+            short: true,
+          },
+          {
+            title: `${workflow} workflow ${status}`,
+            value: `${message}`, // Custom value
+            short: false, // long fields will be full width
+          },
+        ],
+      },
+    ],
+  };
+  return messageBody;
+}
+
+/**
+ * Handles the actual sending request.
+ * We're turning the https.request into a promise here for convenience
+ *
+ * @param webhookURL URL of the Slack channel Webhook
+ * @param messageBody Payload to send to Slack
+ */
+async function sendSlackMessage(
+  webhookURL: string,
+  messageBody: Record<string, any>,
```

Comment:
Seems like you know what the shape of this `messageBody` is based on the return of `prepareSlackNotification` so you could create an interface for that so you don't need to use `any`. If you want to keep this generic because you see in the future then maybe instead of `any`, we can specify values that are acceptable? Would something like this work `Record<string, string | number | boolean | object | array>`?